### PR TITLE
Testing

### LINKED ISSUE #238: Swap enzyme for react-testing-library
[See this talk](https://youtu.be/2HnNo4t8534) for background

### PR DESCRIPTION
**Pull request by dchao19**
This PR closes #238 and #208 

#### What does this PR do?
This PR switches enzyme for react-testing-library and updates the small but existing tests to use the new library. It adds mocks about user authentication and authorization, and adds new tests for the Header component.

### CODE REVIEW

File: src/__tests__/routing.test.js [hunk 1]
```diff
@@ -1,15 +1,21 @@
 import React from 'react';
 import { mountWithContext } from '../testUtils';
+import { cleanup } from 'react-testing-library';
 import App from '../App';
+import { mockUnauthenticated } from '../mocks/localStorageMock';
+
+afterEach(cleanup);
```
Comment (wdoug) [inline src/__tests__/routing.test.js:7]: Huh, I didn't know you could run `afterEach` outside of a `describe` block

Comment (dchao19) [reply src/__tests__/routing.test.js:7]: Frankly, I'm not actually sure it's doing anything, but it also doesn't hurt?

File: .vscode/launch.json [hunk 2]
```diff
@@ -1,34 +1,30 @@
...
-        ],
-        "cwd": "${workspaceRoot}",
-        "protocol": "inspector",
-        "console": "integratedTerminal",
-        "internalConsoleOptions": "neverOpen"
-        }
-    ]
-}
\ No newline at end of file
+  // Use IntelliSense to learn about possible attributes.
```
Comment (dchao19) [inline .vscode/launch.json]: Ooh, didn't realize this got in here. I'm going to revert the changes to this file.

File: src/testUtils.js [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
 import { MemoryRouter } from 'react-router-dom';
 import { MockedProvider } from 'react-apollo/test-utils';
+import withLoggedInUserMock from './mocks/withLoggedInUserMock';
 
-export function mountWithContext(cmp, { routes = [] }) {
-  return mount(
-    <MockedProvider>
+export function mountWithContext(cmp, { routes = ['/'] } = {}, mocks) {
+  return render(
+    <MockedProvider mocks={mocks} removeTypename addTypename={false}>
```
Comment (wdoug) [inline src/testUtils.js:9]: What does `addTypename` do?

File: src/setupTests.js [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
-configure({ adapter: new Adapter() });
+import * as withLoggedInUser from './utils/withLoggedInUser';
 
 getEnvironmentVariableModule.default = jest.fn().mockReturnValue({
   auth0ClientId: '',
   auth0Domain: '',
   graphcoolApi: ''
 });
+
+withLoggedInUser.isAuthenticated = jest.fn().mockReturnValue(true);
```
Comment (wdoug) [inline src/setupTests.js]: Do we need these with the localStorage mocks?

File: src/utils/withAuthSession.js [hunk 5]
```diff
@@ -40,7 +42,7 @@ function logout() {
   localStorage.removeItem(BEARER_TOKEN);
 }
 
-function isAuthenticated() {
+export function isAuthenticated() {
```
Comment (wdoug) [inline src/utils/withAuthSession.js]: Are these exported versions actually used?

File: src/utils/withAuthSession.js [hunk 6]
```diff
@@ -51,7 +53,7 @@ function isAuthenticated() {
   return authenticated;
 }
 
-function getUpdatedState() {
+export function getUpdatedState() {
```
Comment (wdoug) [inline src/utils/withAuthSession.js]: Same question ^

File: src/mocks/localStorageMock.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  getItem: jest.fn(key => null),
+  setItem: jest.fn(),
+  removeItem: jest.fn(),
+  clear: jest.fn()
+};
+
+export const adminUser = {
+  ...unauthenticated,
+  getItem: jest.fn(key => {
+    switch (key) {
```
Comment (wdoug) [inline src/mocks/localStorageMock.js]: Instead of a switch statement here, how do you feel about defining an object and just grabbing the key in the object? i.e.
```js
jest.fn(key => {
  const adminLocalStorageData = {
    ACCESS_TOKEN_KEY: 'accessToken',
    EXPIRES_AT_KEY: '2535419869506',
    USER_ID: adminUserId,
    BEARER_TOKEN: 'bearerToken',
  };
  return adminLocalStorageData[key];
})
```

File: src/testUtils.js (hunk 3 above)
Comment (dchao19) [reply src/testUtils.js:9]: Some Apollo queries add an extra field `__typename` but the mocks don't support this property and will case the mock to not match a query. See: https://www.apollographql.com/docs/guides/testing-react-components.html#addTypename

File: src/testUtils.js [hunk 8]
```diff
@@ -1,12 +1,24 @@
...
 }
+
+export function mountWithAuth(
+  cmp,
+  { routes = ['/'] } = {},
+  additionalMocks = []
+) {
+  return mountWithContext(cmp, routes, [
+    ...additionalMocks,
+    ...withLoggedInUserMock
```
Comment (wdoug) [inline src/testUtils.js]: I would have thought this would not be an array and would just be `withLoggedInUserMock` an object to be consistent with the other exported object in that file

**Review by wdoug — APPROVED** [review]
:shipit:
